### PR TITLE
Clean up query code

### PR DESCRIPTION
**Pull request by dstansby**
As suggested by @ebuchlin in https://github.com/sunpy/sunpy-soar/issues/32#issuecomment-1225915677

### CODE REVIEW

File: sunpy_soar/client.py [hunk 1]
```diff
@@ -84,10 +78,13 @@ def _do_search(query):
         astropy.table.QTable
             Query results.
         """
-        url = SOARClient._construct_url(query)
-        log.debug(f'Getting request from URL: {url}')
+        tap_endpoint = 'http://soar.esac.esa.int/soar-sl-tap/tap'
+        payload = SOARClient._construct_payload(query)
+        # Need to force requests to not form-encode the paramaters
+        payload = '&'.join([f'{key}={val}' for key, val in payload.items()])
```
Comment (ebuchlin) [inline sunpy_soar/client.py:84]: Ah, if you need this, doesn't this defeat the purpose of having the payload as a dict? What is form-encoded that should not be form-encoded? I guess the '+'s from the ADQL query, but these could be spaces in the first place (then encoded to '+'s; for example, I have code in which the ADQL query is built by `f"SELECT filename FROM v_sc_data_item WHERE (instrument='SPICE') AND (insertion_time>='{date_range_str[0]}') AND (insertion_time<='{date_range_str[1]}')"`). Anything else?

Comment (dstansby) [reply sunpy_soar/client.py:84]: Yeah, unfortunately I think spaces or + characters are form encoded, so neither works. I think this PR is still an improvement though, so thanks for suggesting :)

**Review by nabobalis — APPROVED** [review]

File: sunpy_soar/client.py (hunk 1 above)
Comment (Cadair) [reply sunpy_soar/client.py:84]: I think using `urllib.parse.urlencode` rather than doing this manually would be a small improvement.

**Comment by dstansby** [comment]
Given this has an approval and I think it's an improvement, I'm going to merge - if anyone wants to improve further, please open a new PR 😃

**Pull request merged**